Carbon.Windows/Functions/New-CEventLog.ps1

```diff
@@ -0,0 +1,48 @@
+
+function New-CEventLog
+{
+    <#
+    .SYNOPSIS
+    Creates an event log.
+
+    .DESCRIPTION
+    The `New-CEventLog` is a drop-in replacement for the `New-EventLog` function that is available in Windows
+    PowerShell, but with support for PowerShell.
+
+    This function will write an error if the event log source already exists.
+
+    .EXAMPLE
+    New-CEventLog -LogName 'TestApp' -Source 'TestLog'
+
+    Demonstrates creating a `TestLog` event log and registers `TestApp` as a source for the log.
+    #>
+    [CmdletBinding(SupportsShouldProcess)]
+    param(
+        # The name of the log to be created.
+        [Parameter(Mandatory)]
+        [String] $LogName,
+
+        # The source of the events to be written to the log.
+        [Parameter(Mandatory)]
+        [String[]] $Source
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
+
+    foreach ($s in $Source)
+    {
+        if ([Diagnostics.EventLog]::SourceExists($s))

```

Comment:
Maybe create a `Test-CEventLog` function for this that can be used by the install/uninstall functions?